resource-detection/src/system/hostname.rs

```diff
@@ -8,9 +8,15 @@ use std::ffi::OsString;
 pub struct HostnameGetter {}
 
 impl HostnameGetter {
-    #[cfg(unix)]
+    #[cfg(target_family = "unix")]
```

Comment:
Definitely for a future iteration, we might also want to check if we should unify this for `agent-control` and `resource-detection` crates.

Reply:
I'll check if it makes sense to create a new task for it